### PR TITLE
feat: search highlights 

### PR DESCRIPTION
**Pull request by JanAckermann**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/9295
## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-app-files/src/search/sdk/list.ts [hunk 1]
```diff
@@ -27,6 +37,27 @@ export default class List implements SearchList {
...
+      if (index !== -1) {
+        const previousWords = words.slice(index - 2, index)
+        const nextWords = words.slice(index + 1, index + 3)
+        newSentence = `${previousWords.join(' ')} <em>${term}</em> ${nextWords.join(' ')}`
+      }
+
+      return newSentence
+    })
+  }
+
```
Comment (JanAckermann) [inline packages/web-app-files/src/search/sdk/list.ts:60]: not needed, just for demonstration purposes

File: packages/web-app-files/src/search/sdk/list.ts [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 import { Component, computed, Ref, unref } from 'vue'
 import { DavProperties, DavProperty } from 'web-client/src/webdav/constants'
 import { Store } from 'vuex'
 
 export const searchLimit = 200
+export const maxHighlightItems = 5
+
+export interface SearchResource extends Resource {
+  highlights?: string[]
+}
```
Comment (JanAckermann) [inline packages/web-app-files/src/search/sdk/list.ts:19]: IDK if this is necessary

File: packages/web-app-files/src/components/Search/List.vue [hunk 3]
```diff
@@ -61,10 +61,21 @@
           :is-selectable="true"
           :sort-by="sortBy"
           :sort-dir="sortDir"
+          :fields-displayed="['name', 'indicators', 'size', 'tags', 'mdate']"
```
Comment (JanAckermann) [inline packages/web-app-files/src/components/Search/List.vue:64]: used to be buggy, as space stuff like quota was shown

File: package.json [hunk 4]
```diff
@@ -83,7 +83,7 @@
     "cucumber-html-reporter": "5.5.1",
     "depcheck": "1.4.3",
     "ejs": "3.1.9",
-    "eslint": "8.40.0",
+    "eslint": "8.43.0",
```
Comment (JanAckermann) [inline package.json:86]: I bumped eslint as locally calling eslint --fix tried always to alter packages/web-pkg/src/composables/router/types.ts
which resulted in a linter error in the ci, which was annoying

File: packages/web-client/src/webdav/constants/dav.ts [hunk 5]
```diff
@@ -28,6 +28,7 @@ export abstract class DavProperty {
   static readonly MimeType: string = '{DAV:}getcontenttype'
   static readonly ResourceType: string = '{DAV:}resourcetype'
   static readonly DownloadURL: string = '{http://owncloud.org/ns}downloadURL'
+  static readonly Highlights: string = '{http://owncloud.org/ns}preview'
```
Comment (JanAckermann) [inline packages/web-client/src/webdav/constants/dav.ts:31]:
```suggestion
  static readonly Highlights: string = '{http://owncloud.org/ns}highlights'
```
as soon changed in backend

File: changelog/unreleased/enhancement-search-list-add-highligthed-file-content [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Search list add highlighted file content
+
+We've introduced a feature that shows a highlighted result summary matching the search term in the search list,
+if there are occurrences in the fount files.
```
Comment (JammingBen) [inline changelog/unreleased/enhancement-search-list-add-highligthed-file-content:4]:
```suggestion
if there are occurrences in the found files.
```

**Review by JammingBen — COMMENTED** [review]
Small typo, rest LGTM 👍

**Review by JammingBen — APPROVED** [review]

**Comment by JanAckermann** [comment]
DO NOT MERGE

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS-1</strong> https://drone.owncloud.com/owncloud/web/36834/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36834/tracing/search-for-content-of-file-alice-2023-6-28-08-30-21.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36834/tracing/search-for-content-of-file-brian-2023-6-28-08-30-25.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36834/tracing/search-in-personal-spaces-alice-2023-6-28-08-34-55.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36834/tracing/search-in-personal-spaces-brian-2023-6-28-08-34-48.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

**Pull request merged**